### PR TITLE
docker-compose 3.7 with x-defaults

### PR DESCRIPTION
**Pull request by daminuxfork**
Add x-default config

### CODE REVIEW

**Review by fzipi — APPROVED** [review]
Thanks, good addition!

File: docker-compose.yaml [hunk 1]
```diff
@@ -7,76 +7,70 @@
...
-  crs-apache:
-    image: owasp/modsecurity-crs:apache
-    ports:
-      - "80:80"
-      # only available if SETTLS was enabled:
-      - "443:443"
-
-    environment:
-      - SERVERNAME=localhost
+version: '3.7'
```
Comment (theseion) [inline docker-compose.yaml:10]: The `version` field is only informative. It's removal has been recommended for some time now: https://docs.docker.com/compose/compose-file/04-version-and-name/.

File: docker-compose.yaml [hunk 2]
```diff
@@ -7,76 +7,70 @@
...
-      #- ALLOWED_REQUEST_CONTENT_TYPE_CHARSET='utf-8|iso-8859-1|iso-8859-15|windows-1252'
-      #- ALLOWED_HTTP_VERSIONS=HTTP/1.0 HTTP/1.1 HTTP/2 HTTP/2.0
-      #- RESTRICTED_EXTENSIONS=.asa/ .asax/ .ascx/ .axd/ .backup/ .bak/ .bat/ .cdx/ .cer/ .cfg/ .cmd/ .com/ .config/ .conf/ .cs/ .csproj/ .csr/ .dat/ .db/ .dbf/ .dll/ .dos/ .htr/ .htw/ .ida/ .idc/ .idq/ .inc/ .ini/ .key/ .licx/ .lnk/ .log/ .mdb/ .old/ .pass/ .pdb/ .pol/ .printer/ .pwd/ .rdb/ .resources/ .resx/ .sql/ .swp/ .sys/ .vb/ .vbs/ .vbproj/ .vsdisco/ .webinfo/ .xsd/ .xsx/
-      #- RESTRICTED_HEADERS=/accept-charset/ /content-encoding/ /proxy/ /lock-token/ /content-range/ /if/
-      #- STATIC_EXTENSIONS=/.jpg/ /.jpeg/ /.png/ /.gif/ /.js/ /.css/ /.ico/ /.svg/ /.webp/
+    #######################################################
+    # Various CRS Variables with Default Values
+    #######################################################
+    #ENFORCE_BODYPROC_URLENCODED: 1
```
Comment (theseion) [inline docker-compose.yaml:49]: Missing space after the comment marker.

File: docker-compose.yaml [hunk 3]
```diff
@@ -7,76 +7,70 @@
...
-      #- MAX_NUM_ARGS=255
-      #- ARG_NAME_LENGTH=100
-      #- ARG_LENGTH=400
-      #- TOTAL_ARG_LENGTH=64000
-      #- MAX_FILE_SIZE=1048576
-      #- COMBINED_FILE_SIZES=1048576
+    #######################################################
+    # CRS Variables with Default Value unlimited
+    #######################################################
+    #MAX_NUM_ARGS: 255
```
Comment (theseion) [inline docker-compose.yaml:61]: Missing space after the comment marker.

File: docker-compose.yaml [hunk 4]
```diff
@@ -86,18 +80,19 @@ services:
...
+      # only available if SETTLS was enabled:      
+      - "443:443"
+    <<: *default-settings
+
   crs-nginx:
-    extends: crs-apache
     image: owasp/modsecurity-crs:nginx
     ports:
-      - "80:8080"
+      - "80:80"
```
Comment (theseion) [inline docker-compose.yaml:95]: That port mapping is intentional ;)
```suggestion
      - "80:8080"
```

**Review by theseion — CHANGES_REQUESTED** [review]

**Review by theseion — APPROVED** [review]

**Comment by theseion** [comment]
Thanks @daminuxfork!

**Pull request merged**